### PR TITLE
Docs deployment concepts

### PR DESCRIPTION
**Pull request by marwan116**
This PR includes a very small documentation fix for the deployment concepts documentation page.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [x] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

**Comment by billpalombi** [comment]
Good catch @marwan116.

You're correct that `image_tag` should be just `tag`, but I believe `image_name` is correct.

**Review by billpalombi — CHANGES_REQUESTED** [review]

File: docs/concepts/deployments.md [hunk 1]
```diff
@@ -422,7 +422,7 @@ deployments:
             cluster_config: "{{ prefect.blocks.kubernetes-cluster-config.my-favorite-config }}"
 ```
 
-So long as our `build` steps produce fields called `image_name` and `image_tag`, every time we deploy a new version of our deployment these fields will be dynamically populated with the relevant values.
+So long as our `build` steps produce fields called `image` and `tag`, every time we deploy a new version of our deployment these fields will be dynamically populated with the relevant values.
```
[the remark this answers is not in the document]
Comment (marwan116) [reply docs/concepts/deployments.md:425]: so {{build-image.image}} is what is used as a job_variable, it is what I understood to be a "produced field" in that it takes the image name and tag and produces the full image reference

[the remark this answers is not in the document]
Comment (billpalombi) [reply docs/concepts/deployments.md:425]: I see. Your interpretation makes sense. This entire sentence is not clearly written. I'll rephrase to make it clearer.

**Review by billpalombi — APPROVED** [review]
Merging and following up with other changes in a seperate PR

**Pull request closed without merging**